src/groupByKey.ts

```diff
@@ -2,6 +2,23 @@
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+  const arrValues: string[] = [];
+
+  for (const item of items) {
+    const groupKey = item[key] as string;
+
+    if (!arrValues.includes(groupKey)) {
+      arrValues.push(groupKey);
+    }
```

Comment:
Instead of checking and adding to `arrValues`, you can directly initialize an empty array in `result` for the `groupKey` if it doesn't exist.